meta-ros2-humble/conf/layer.conf

```diff
@@ -0,0 +1,26 @@
+# We have a conf and classes directory, append to BBPATH
+BBPATH .= ":${LAYERDIR}"
+
+# We have a recipes directory, add to BBFILES
+BBFILES += "${LAYERDIR}/recipes-*/*/*.bb ${LAYERDIR}/recipes-*/*/*.bbappend ${LAYERDIR}/generated-recipes/*/*.bb"
+
+BBFILE_COLLECTIONS += "ros2-humble-layer"
+BBFILE_PATTERN_ros2-humble-layer := "^${LAYERDIR}/"
+BBFILE_PRIORITY_ros2-humble-layer = "12"
```

Comment:
All the ROS distro layers are 12 so that the common ROS layers have higher priority:

$ find . -name layer.conf | xargs grep ^BBFILE_PRIORITY
**./meta-ros-common/conf/layer.conf:BBFILE_PRIORITY_ros-common-layer = "10"**
**./meta-ros2/conf/layer.conf:BBFILE_PRIORITY_ros2-layer = "11"**
./meta-ros2-humble/conf/layer.conf:BBFILE_PRIORITY_ros2-humble-layer = "12"
./meta-ros2-foxy/conf/layer.conf:BBFILE_PRIORITY_ros2-foxy-layer = "12"
**./meta-ros-python2/conf/layer.conf:BBFILE_PRIORITY_ros-python2 = "9"**
./meta-ros1-noetic/conf/layer.conf:BBFILE_PRIORITY_ros1-noetic-layer = "12"
./meta-ros1-melodic/conf/layer.conf:BBFILE_PRIORITY_ros1-melodic-layer = "12"
./meta-ros2-galactic/conf/layer.conf:BBFILE_PRIORITY_ros2-galactic-layer = "12"
**./meta-ros1/conf/layer.conf:BBFILE_PRIORITY_ros1-layer = "11"**
./meta-ros2-dashing/conf/layer.conf:BBFILE_PRIORITY_ros2-dashing-layer = "12"
./meta-ros2-eloquent/conf/layer.conf:BBFILE_PRIORITY_ros2-eloquent-layer = "12"
./meta-ros2-rolling/conf/layer.conf:BBFILE_PRIORITY_ros2-rolling-layer = "12"

The common ROS layers start around 10 to make room for the common layers in OpenEmbedded:
$ bitbake-layers show-layers
NOTE: Starting bitbake server...
layer                 path                                      priority
==========================================================================
meta-raspberrypi      /ala-lpggp31/rwoolley/pr1035/meta-raspberrypi  9
meta-ros2-humble      /ala-lpggp31/rwoolley/pr1035/meta-ros/meta-ros2-humble  12
meta-ros2             /ala-lpggp31/rwoolley/pr1035/meta-ros/meta-ros2  11
meta-ros-common       /ala-lpggp31/rwoolley/pr1035/meta-ros/meta-ros-common  10
meta-python           /ala-lpggp31/rwoolley/pr1035/meta-openembedded/meta-python  7
meta-oe               /ala-lpggp31/rwoolley/pr1035/meta-openembedded/meta-oe  6
meta                  /ala-lpggp31/rwoolley/pr1035/openembedded-core/meta  5